packages/wdio-utils/src/envDetector.ts

```diff
@@ -106,6 +106,15 @@ function isMobile(capabilities: WebdriverIO.Capabilities) {
     const browserName = (capabilities.browserName || '').toLowerCase()
     const bsOptions = capabilities['bstack:options'] || {}
     const browserstackBrowserName = (bsOptions.browserName || '').toLowerCase()
+
+    /**
+     * it is _not_ a mobile session if automationName is set to start a mobile session
+     */
+    const automationName = capabilities['appium:options']?.automationName || capabilities['appium:automationName']
```

Comment:
@christian-bromann 

this is strange, we are automating a mobile device, but only, if I understand it correctly, with a "browser driver". Shouldn't we introduce a new flag for this and use that to exclude specific mobile commands for non-mobile browser drivers?

Reply:
> we are automating a mobile device

No, we are using Appium to automate a browser.

> Shouldn't we introduce a new flag for this and use that to exclude specific mobile commands for non-mobile browser drivers?

I think there is a difference here, users can:

- run a browser session on a mobile device: WebdriverIO properly detects this and sets `isMobile` to true (which I am not sure is right), however Appium in this case supports `getContext` (I think)
- we are using Appium to automate a desktop browser, here `isMobile` should be false but was detected as `true`

This patch fixes the second case. I would need more investigation to better understand the first case.

Reply:
@christian-bromann 

This [link/config](https://github.com/appium/appium-geckodriver/blob/master/test/functional/mobile-driver-e2e-specs.js) from @mykola-mokhnach suggests something else. This is FF on Android which would still be a mobile session.

> run a browser session on a mobile device: WebdriverIO properly detects this and sets isMobile to true (which I am not sure is right), however Appium in this case supports getContext (I think)
From what I know the default drivers (appium-xcuitest-driver and appium-uiautomator2-driver) can automate both, meaning web and native. That's the reason why they support the `getContext`.

For now we can proceed with this change, but then we need to add a comment to the code that we need to rethink this logic